src/test/ui/proc-macro/mixed-site-span.stderr

```diff
@@ -2,19 +2,13 @@ error[E0426]: use of undeclared label `'label_use`
   --> $DIR/mixed-site-span.rs:15:9
    |
 LL |         proc_macro_rules!();
-   |         ^^^^^^^^^^^^^^^^^^^^
-   |         |
-   |         undeclared label `'label_use`
-   |         in this macro invocation
```

Comment:
Not sure why "in this macro invocation" messages are going away.

Reply:
Finally figured it out, "in this macro invocation" should be skipped when it's on the same span as the message itself.

Why can that happen? Well, it's *specifically* because we replace spans inside external macros (and after this PR inside any cross-crate-imported files) with the macro call site, so that we don't show implementation details.

If the span would be identical, the message is redundant since it doesn't help anything. But the logic is confusing because this is not explicit anywhere, it just happens to check the same thing in multiple places!

Hopefully I can clean this up and comment it a bit.